master/internal/experiment/postgres_experiments.go

```diff
@@ -46,3 +54,131 @@ func GetExperimentAndCheckCanDoActions(
 	}
 	return e, *curUser, nil
 }
+
+// ExperimentCheckpointsToGCRaw returns a comma-separated string describing checkpoints
+// that should be GCed according to the given GC policy parameters. If the delete parameter is true,
+// the returned checkpoints are also marked as deleted in the database.
+func ExperimentCheckpointsToGCRaw(
+	ctx context.Context,
+	id int,
+	experimentBest, trialBest, trialLatest int,
+) ([]uuid.UUID, error) {
+	// The string for the CTEs that we need whether or not we're not deleting the results. The
+	// "selected_checkpoints" table contains the checkpoints to return as rows, so that we can easily
+	// set the corresponding checkpoints to deleted in a separate CTE if we're deleting.
+	query := `
+WITH const AS (
```

Comment:
nit (non-blocking): this might be a case converting this out of the raw SQL and using bun to build the query could be improve readability; it could be more of a hassle than it's worth, so take this with a grain of salt

Reply:
This particular query has been quite buggy in the past and caused alot of issues. I'm not sure it's worth messing with incase we cause bugs again - especially not in this PR. We could create a ticket to bunify it seperatly

Reply:
+1 to creating a separate ticket for it